internal/component/database_observability/mysql/component.go

```diff
@@ -206,6 +208,33 @@ func (c *Component) Update(args component.Arguments) error {
 	return nil
 }
 
+func enableOrDisableCollectors(a Arguments) map[string]bool {
+	// configurable collectors and their default enabled/disabled value
+	collectors := map[string]bool{
+		querySample: true,
+		schemaTable: true,
+	}
+
+	for _, disabled := range a.DisableCollectors {
+		for c := range collectors {
+			if c == disabled {
+				collectors[c] = false
+				break
+			}
+		}
```

Comment:
Instead of looping, can we check directly with?

```
		if _, ok := collectors[disabled]; ok {
			collectors[disabled] = false
		}
```

Reply:
good point, thanks